addons/account/models/account_bank_statement.py

```diff
@@ -400,530 +441,609 @@ def button_journal_entries(self):
-    account_id = fields.Many2one('account.account', string='Counterpart Account', domain=[('deprecated', '=', False)],
-        help="This technical field can be used at the statement line creation/import time in order to avoid the reconciliation"
-             " process on it later on. The statement line will simply create a counterpart on this account")
-    statement_id = fields.Many2one('account.bank.statement', string='Statement', index=True, required=True, ondelete='cascade')
-    journal_id = fields.Many2one('account.journal', related='statement_id.journal_id', string='Journal', store=True, readonly=True)
-    partner_name = fields.Char(help="This field is used to record the third party name when importing bank statement in electronic format,"
-             " when the partner doesn't exist yet in the database (or cannot be found).")
-    ref = fields.Char(string='Reference')
-    note = fields.Text(string='Notes')
+    partner_name = fields.Char(help="This field is used to record the third party name when importing bank statement in electronic format, when the partner doesn't exist yet in the database (or cannot be found).")
     transaction_type = fields.Char(string='Transaction Type')
-    sequence = fields.Integer(index=True, help="Gives the sequence order when displaying a list of bank statement lines.", default=1)
-    company_id = fields.Many2one('res.company', related='statement_id.company_id', string='Company', store=True, readonly=True)
-    journal_entry_ids = fields.One2many('account.move.line', 'statement_line_id', 'Journal Items', copy=False, readonly=True)
-    amount_currency = fields.Monetary(help="The amount expressed in an optional other currency if it is a multi-currency entry.")
-    currency_id = fields.Many2one('res.currency', string='Currency', help="The optional other currency if it is a multi-currency entry.")
+    payment_ref = fields.Char(string='Label', store=True, required=True,
+        compute='_compute_from_move_lines',
+        inverse='_inverse_payment_ref')
+    amount = fields.Monetary(currency_field='currency_id', store=True,
+        compute='_compute_from_move_lines',
+        inverse='_inverse_amounts')
+    amount_currency = fields.Monetary(currency_field='foreign_currency_id', store=True,
+        compute='_compute_from_move_lines',
+        inverse='_inverse_amounts',
+        help="The amount expressed in an optional other currency if it is a multi-currency entry.")
+    foreign_currency_id = fields.Many2one('res.currency', string='Foreign Currency', store=True,
+        compute='_compute_from_move_lines',
+        inverse='_inverse_amounts',
+        help="The optional other currency if it is a multi-currency entry.")
+    partner_id = fields.Many2one('res.partner', string='Partner', ondelete='restrict', store=True,
+        compute='_compute_from_move_lines',
+        inverse='_inverse_partner_id',
```

Comment:
So what? I need to set the commercial partner here. However, `commercial_partner_id` is a computed field on `account.move` so I can't reuse it. I tried to reuse one existing field of `account.move` without any success...

Reply:
that's a real pity to have that field defined in both table, that's really confusing!

* add a comment: field is defined in both tables, we don't change its name because it would be a mess to track/fix
* also add a TODO: remove that field here, whenever the compute-readonly-false-inverse-whatsoever can be defined on account.move directly